projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -101,7 +101,18 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
      * ```
      */
     @Input()
-    public igxForItemSize: any;
+    get igxForItemSize(): any {
```

Comment:
Input prop changes are handled in ngOnChanges. initSizesCache is already called there in case igxForItemSize changes. You can move the additional code there.

Reply:
Actually I did it this way initially, but this will require having input binding for the directive to change the property. For example [this](https://github.com/IgniteUI/igniteui-angular/blob/master/projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts#L458) won't trigger the change detection and won't change the item size.

Reply:
Ok, ignore my previous comment, I discussed it with @ViktorSlavov  and @damyanpetev and I'm reverting it back to ngOnChanges